java/arcs/core/host/ParticleContext.kt

```diff
@@ -52,7 +52,15 @@ class ParticleContext(
     private val desyncedHandles: MutableSet<Handle> = mutableSetOf()
 
     // One-shot callback used to notify the arc host that the particle is in the Running state.
+    // If the particle is already in the running state, the method will be called, but not set.
     private var notifyReady: ((Particle) -> Unit)? = null
+        set(value) {
+            if (particleState == ParticleState.Running) {
+                value?.invoke(this.particle)
+            } else {
+                field = value
```

Comment:
for it to be a one-shot, should it wrap the `value` in a new method which re-sets `notifyReady` to null after the wrapper is called?

E.g.

```kotlin
private var notifyReady: ((Particle) -> Unit)? = null
    set(value) {
        if (particleState == ParticleState.Running) {
            value?.invoke(this.particle)
        } else if (value != null) {
            field = { particle ->
                value()
                notifyReady = null
            }
        }
    }
```

Reply:
I considered that, and then decided against it in the name of keeping this PR simple. It seems like a reasonable suggestion, but it also seems like one more thing that can somehow surprisingly go wrong.

Reply:
(also, the value is currently cleared in the `moveToReady` function which uses this)